waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -913,23 +913,54 @@ proc addAll*(wakuRlnRelay: WakuRLNRelay, list: seq[IDCommitment]): RlnRelayResul
+proc generateGroupUpdateHandler(rlnPeer: WakuRLNRelay): GroupUpdateHandler =
+  # assuming all the members arrive in order
+  # TODO: check the index and the pubkey depending on
+  # the group update operation
+  var handler: GroupUpdateHandler
+  handler = proc(pubkey: Uint256, index: Uint256): RlnRelayResult[void] {.raises: [Defect].} =
+    var pk: IDCommitment
     try:
-      debug "onRegister", pubkey = pubkey, index = index
-      handler(pubkey, index)
-    except Exception as err:
-      # chronos still raises exceptions which inherit directly from Exception
-      error "Error handling new member registration: ", err=err.msg
-      doAssert false, err.msg
-  do (err: CatchableError):
-    error "Error from subscription: ", err=err.msg
+      pk = pubkey.toIDCommitment()
+    except:
+      return err("invalid pubkey")
+    let isSuccessful = rlnPeer.insertMember(pk)
+    if isSuccessful.isErr():
+      return err("failed to add a new member to the Merkle tree")
+    else:
+      debug "new member added to the Merkle tree", pubkey=pubkey, index=index
+      debug "acceptable window", validRoots=rlnPeer.validMerkleRoots.mapIt(it.inHex)
+      rlnPeer.lastSeenMembershipIndex = index.toMembershipIndex()
+      return ok()
+  return handler
+
+proc subscribeToMemberRegistrations(web3: Web3, 
+                                    contractAddress: Address,
+                                    fromBlock: string = "0x0",
+                                    handler: GroupUpdateHandler): Future[Subscription] {.async, gcsafe.} =
+  var contractObj = web3.contractSender(MembershipContract, contractAddress)
 
-proc subscribeToGroupEvents(ethClientUri: string, ethAccountAddress: Option[Address] = none(Address), contractAddress: Address, blockNumber: string = "0x0", handler: RegistrationEventHandler) {.async, gcsafe.} = 
+  let onMemberRegistered = proc (pubkey: Uint256, index: Uint256) {.gcsafe.} =
+    debug "onRegister", pubkey = pubkey, index = index
+    let groupUpdateRes = handler(pubkey, index)
+    if groupUpdateRes.isErr():
+      error "Error handling new member registration", err=groupUpdateRes.error()
+
+  let onError = proc (err: CatchableError) =
+    error "Error in subscription", err=err.msg
+
+  return await contractObj.subscribe(MemberRegistered,
+                                     %*{"fromBlock": fromBlock, "address": contractAddress},
+                                     onMemberRegistered,
+                                     onError)
+
+proc subscribeToGroupEvents*(ethClientUri: string,
+                            ethAccountAddress: Option[Address] = none(Address),
+                            contractAddress: Address,
+                            blockNumber: string = "0x0",
+                            handler: GroupUpdateHandler) {.async, gcsafe.} = 
   ## connects to the eth client whose URI is supplied as `ethClientUri`
```

Comment:
Please update the proc description to match the new semantics.

Reply:
Addressed in [582a6c3](https://github.com/status-im/nwaku/pull/1296/commits/582a6c322025d6d125f97ad434311170bb454c26)